Topics/Tech_Stacks/swift.md

```diff
@@ -0,0 +1,73 @@
+
+These are just some of the many advantages available.
+
+## What is SwiftUI?
+In 2019, Apple introduced a new framework for building user interfaces called SwiftUI. This is a [declarative UI toolkit](https://www.hackingwithswift.com/quick-start/swiftui/what-is-swiftui), similar to React, where we can tell it what components we want
+and the framework will handle performing the steps needed to get that result.
+
+[Hacking with Swift](https://www.hackingwithswift.com) is a great free resource for learning how to do specific things in Swift and SwiftUI
+when the issues arise.
+
+## Starting a Swift Project
+After launching Xcode, selecting `Create a new Xcode project`, and choosing which platform and type of app you want to make, you will then have
+to fill in the following info:
+
+<img width="1512" alt="Screen Shot 2023-03-20 at 4 31 06 PM" src="https://user-images.githubusercontent.com/65694861/226458997-0d9c6227-9ce6-4d34-91b0-58c69b02d016.png">
+
+This information can be changed later, so for starters you can leave `Team` to be empty, as this is only necessary for deploying the app to the App Store. `Organization identifier` is used to uniquely identify your app once it is up on the App Store, so you can choose whichever name you'd like, such as your name or group's name. Do note `Organization identifier` cannot be changed once the app is uploaded to the App Store but it is purely meta data.
+
+Make sure to use `SwiftUI` and `Swift` as your interface and language respectively, then click `Next` to choose where to store your project, and now you're ready to start.
+
+## Testing Your App
+Xcode has [built-in simulators for many Apple devices](https://developer.apple.com/documentation/xcode/running-your-app-in-simulator-or-on-a-device)
+you can use to run your code and see how it performs. You can also download new simulators for specific a device and operating system version to test
+different scenarios, such as an iPhone 11 running iOS 13.
+
+If you have your own Apple device, you can also connect it to your Mac device and run your app on it for testing. Note that as of iOS 14, from your device, you will have to first go to `Settings` > `Privacy & Security` > `Developer Mode` first to allow your device to run apps downloaded from Xcode.
+
+## Other Useful Resources
+[Learn about the different data types in Swift](https://www.hackingwithswift.com/read/0/3/types-of-data). Each language has its own nuances in how
+variables are declared and stored, useful to become familiar with it before starting to code.
+
+[SwiftUI has many property wrappers that provide different functionality to the object instances they are attached to](https://www.hackingwithswift.com/quick-start/swiftui/all-swiftui-property-wrappers-explained-and-compared). These are an important tool for making your code efficient and storing your objects properly. In particular, you should be aware of what the `@State` and `@Binding` wrappers do.
```

Comment:
This is okay as is, but it might be nice to mention some parallels between SwiftUI and React. For example, `@State` is equivalent to `useState()` in React.

Reply:
Thinking about this more, It's wrong to assume the reader has a background in React. Resolving.